From e56cb4ebb98fe814a2f88d47d92e480a14be4a72 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Sun, 23 May 2021 22:19:40 +0200
Subject: [PATCH 35/39] cinterion: reuse the same method for load and reload

The load_() method is used for connection monitoring; while the
reload_() method is used to sync connection status after a
suspend/resume operation. The same method can be used for both things
in the Cinterion plugin.
---
 plugins/cinterion/mm-broadband-bearer-cinterion.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/cinterion/mm-broadband-bearer-cinterion.c b/plugins/cinterion/mm-broadband-bearer-cinterion.c
index 464e75e9..861c7ccb 100644
--- a/plugins/cinterion/mm-broadband-bearer-cinterion.c
+++ b/plugins/cinterion/mm-broadband-bearer-cinterion.c
@@ -681,6 +681,10 @@ mm_broadband_bearer_cinterion_class_init (MMBroadbandBearerCinterionClass *klass
 
     base_bearer_class->load_connection_status        = load_connection_status;
     base_bearer_class->load_connection_status_finish = load_connection_status_finish;
+#if defined WITH_SYSTEMD_SUSPEND_RESUME
+    base_bearer_class->reload_connection_status        = load_connection_status;
+    base_bearer_class->reload_connection_status_finish = load_connection_status_finish;
+#endif
 
     broadband_bearer_class->dial_3gpp              = dial_3gpp;
     broadband_bearer_class->dial_3gpp_finish       = dial_3gpp_finish;
-- 
2.31.1

